admin/server/server.go

```diff
@@ -89,7 +88,7 @@ func (s *Server) Serve(ctx context.Context, port int) error {
 		return func(c echo.Context) error {
 			defer func() {
 				if r := recover(); r != nil {
-					if r == http.ErrAbortHandler {
+					if errors.Is(r.(error), http.ErrAbortHandler) {
```

Comment:
Doing `r.(error)` here is unsafe. You must check the type first. You can panic with other values than errors (e.g. `panic("not implemented")` panics with a string)